cin_validator/rules/cin2022_23/rule_8615.py

```diff
@@ -0,0 +1,161 @@
+        "(S47ActualStartDate > DateOfInitialCPC) or (DateOfInitialCPC.notna() and S47ActualStartDate.isna())",
+        inplace=True,
+    )
+
+    df_issues = df.reset_index()
+
+    link_id = tuple(
+        zip(
+            df_issues[LAchildID],
+            df_issues[S47ActualStartDate],
+            df_issues[DateOfInitialCPC],
+        )
+    )
+    df_issues["ERROR_ID"] = link_id
+    df_issues = df_issues.groupby("ERROR_ID")["ROW_ID"].apply(list).reset_index()
+
+    rule_context.push_type_1(
+        table=Section47,
+        columns=[S47ActualStartDate, DateOfInitialCPC],
+        row_df=df_issues,
+    )
+
+
+def test_validate():
+    IDS_are = [
+        "AAAAAAAA",
+        "BBBBBBBBB",
+        "CCCCCCCCCCC",
+        "DDDDDDDDD",
+        "EEEE",
+        "FFFFFFFFF",
+        "GGGGGGGGGG",
+        "HHHH",
+    ]
+    S47start = [
+        "01-01-2020",
+        "01-02-2020",
+        "01-03-2020",
+        "15-01-2020",
+        pd.NA,
+        "01-07-2020",
+        "15-01-2020",
+        pd.NA,
+    ]
+    ICPCstart = [
+        "01-01-2020",
+        "01-01-2020",
+        "01-03-2020",
+        "17-01-2020",
+        pd.NA,
+        "01-01-2020",
+        "15-01-2020",
+        "01-01-2020",
+    ]
+    fake_dataframe = pd.DataFrame(
+        {
+            "LAchildID": IDS_are,
+            "S47ActualStartDate": S47start,
+            "DateOfInitialCPC": ICPCstart,
+        }
+    )
+
+    fake_dataframe["S47ActualStartDate"] = pd.to_datetime(
+        fake_dataframe["S47ActualStartDate"], format=r"%d-%m-%Y", errors="coerce"
+    )
+    fake_dataframe["DateOfInitialCPC"] = pd.to_datetime(
+        fake_dataframe["DateOfInitialCPC"], format=r"%d-%m-%Y", errors="coerce"
+    )
+
+    result = run_rule(validate, {Section47: fake_dataframe})
+
+    issues = result.type1_issues
+
+    issue_table = issues.table
+    assert issue_table == Section47
+
+    issue_columns = issues.columns
+    assert issue_columns == [S47ActualStartDate, DateOfInitialCPC]
+
+    issue_rows = issues.row_df
+
+    assert len(issue_rows) == 3
+
+    assert isinstance(issue_rows, pd.DataFrame)
+    assert issue_rows.columns.to_list() == ["ERROR_ID", "ROW_ID"]
+
+    expected_df = pd.DataFrame(
+        [
+            {
+                "ERROR_ID": (
+                    "BBBBBBBBB",
+                    pd.to_datetime("01-02-2020", format=r"%d-%m-%Y", errors="coerce"),
+                    pd.to_datetime("01-01-2020", format=r"%d-%m-%Y", errors="coerce"),
+                ),
+                "ROW_ID": [1],
+            },
+            {
+                "ERROR_ID": (
+                    "FFFFFFFFF",
+                    pd.to_datetime("01-07-2020", format=r"%d-%m-%Y", errors="coerce"),
+                    pd.to_datetime("01-01-2020", format=r"%d-%m-%Y", errors="coerce"),
+                ),
+                "ROW_ID": [5],
+            },
+            {
+                "ERROR_ID": (
+                    "HHHH",
+                    pd.NaT,
```

Comment:
It took me 2 hours to discover there's a difference between pd.NA  &   pd.NaT

Reply:
This was a helpful titbit for me when writing the test on another rule today. Thanks!